Sources/NIO/SocketChannel.swift

```diff
@@ -12,6 +12,14 @@
 //
 //===----------------------------------------------------------------------===//
 
+#if os(macOS) || os(iOS) || os(watchOS) || os(tvOS)
+import CNIODarwin
+#elseif os(Linux) || os(FreeBSD) || os(Android)
+import CNIOLinux
+#else
+let badOS = { fatalError("unsupported OS") }()
+#endif
```

Comment:
Do we need this? I think it'd be better to fatal error only if you try to _use_ the new functionality.